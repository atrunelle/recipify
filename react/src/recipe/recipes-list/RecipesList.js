import React, { Component } from 'react';
import { connect } from 'react-redux';

class RecipesList extends Component {
  render() {
    if (!this.props.recipes.length) return (<div>
      <p>You don't have any recipes yet</p>
      <a>Create recipes</a>
    </div>);
    return (
      <div></div>
    );
  }
}

const mapStateToProps = (state) => ({
	recipes: state.recipes.recipes,
});

export default connect(mapStateToProps)(RecipesList);