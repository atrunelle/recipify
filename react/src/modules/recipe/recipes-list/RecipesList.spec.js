import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import createRouterContext from 'react-router-test-context';
import PropTypes from 'prop-types';

import RecipesList from './RecipesList';

describe('Component: RecipesList', () => {
  const initialState = {
    recipes: {
      recipes: [],
    },
  };
  const mockStore = configureStore();
  let store;
  const context = createRouterContext()
  const childContextTypes = {
    router: PropTypes.object,
  };

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('should match snapshot', () => {
    const component = mount(
      <Provider store={store}>
        <RecipesList />
      </Provider>,
      { context, childContextTypes },
    );
    expect(component.html()).toMatchSnapshot();
  });
});