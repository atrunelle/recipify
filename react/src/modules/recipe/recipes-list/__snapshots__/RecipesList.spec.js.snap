// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RecipesList should match snapshot 1`] = `"<div class=\\"u-push-top\\"><div class=\\"u-text-center\\"><div class=\\"circle-icon\\"><img class=\\"circle-icon__icon\\" src=\\"recipes.svg\\" alt=\\"recipes icon\\"></div><h2 class=\\"MuiTypography-root-1 MuiTypography-headline-6 MuiTypography-gutterBottom-18\\">No recipes yet</h2><p class=\\"MuiTypography-root-1 MuiTypography-body1-10 MuiTypography-gutterBottom-18 MuiTypography-paragraph-19\\">Do not forget to save recipes when you play with ingredients. Once saved, they will show here.</p><a tabindex=\\"0\\" class=\\"MuiButtonBase-root-41 MuiButton-root-25 MuiButton-raised-31\\" role=\\"button\\" href=\\"/\\"><span class=\\"MuiButton-label-26\\">Create recipe</span><span class=\\"MuiTouchRipple-root-44\\"></span></a></div></div>"`;
